### PR TITLE
search-react,techdocs: remove core-app-api dependency

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This needs a lint rule/check 😅 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-react/package.json [hunk 1]
```diff
@@ -43,6 +42,7 @@
     "react": "^16.13.1 || ^17.0.0"
   },
   "devDependencies": {
+    "@backstage/core-app-api": "^1.0.1",
```
Comment (freben) [inline plugins/search-react/package.json:45]: can't storybook use the TestApiProvider? 🤔  isn't that the only active use

**Review by freben — APPROVED** [review]

File: plugins/search-react/package.json (hunk 1 above)
Comment (Rugvip) [reply plugins/search-react/package.json:45]: Either or really, not too big of a deal as long as `core-app-api` is a dev dep

**Comment by Rugvip** [comment]
@backstage/techdocs-core good to go?

File: plugins/techdocs/src/Router.tsx [hunk 2]
```diff
@@ -69,10 +68,10 @@ export const EmbeddedDocsRouter = (props: PropsWithChildren<{}>) => {
   }
 
   return (
-    <FlatRoutes>
```
Comment (iamEAP) [inline plugins/techdocs/src/Router.tsx:72]: Hmmmmm.  This breaks Addons on the entity page.

Comment (iamEAP) [reply plugins/techdocs/src/Router.tsx:72]: Here's a diff you can apply to this branch or master for illustration purposes:

```diff
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 26068c6c8f..a0a5d87c77 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -138,6 +138,8 @@ import {
 import { EntityGoCdContent, isGoCdAvailable } from '@backstage/plugin-gocd';
 
 import React, { ReactNode, useMemo, useState } from 'react';
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { TextSize } from '../techdocs/TextSize';
 
 const customEntityFilterKind = ['Component', 'API', 'System'];
 
@@ -397,7 +399,11 @@ const serviceEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
 
     <EntityLayout.Route
@@ -464,7 +470,11 @@ const websiteEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
     <EntityLayout.Route
       if={isNewRelicDashboardAvailable}
@@ -503,7 +513,11 @@ const defaultEntityPage = (
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      <EntityTechdocsContent>
+        <TechDocsAddons>
+          <TextSize />
+        </TechDocsAddons>
+      </EntityTechdocsContent>
     </EntityLayout.Route>
 
     <EntityLayout.Route path="/todos" title="TODOs">
diff --git a/packages/app/src/components/techdocs/TextSize.tsx b/packages/app/src/components/techdocs/TextSize.tsx
new file mode 100644
index 0000000000..9c410dd052
--- /dev/null
+++ b/packages/app/src/components/techdocs/TextSize.tsx
@@ -0,0 +1,51 @@
+import React, { useEffect, useCallback, useState } from 'react';
+import { Tooltip, ButtonGroup, Button } from '@material-ui/core';
+import ZoomInIcon from '@material-ui/icons/ZoomIn';
+import ZoomOutIcon from '@material-ui/icons/ZoomOut';
+
+import { createTechDocsAddonExtension, TechDocsAddonLocations, useShadowRoot } from '@backstage/plugin-techdocs-react';
+import { techdocsPlugin } from '@backstage/plugin-techdocs';
+
+const TYPESET = '.md-typeset';
+
+export const TextSizeAddon = () => {
+  const shadowRoot = useShadowRoot();
+  const [fontSize, setFontSize] = useState<number>(16);
+
+  useEffect(() => {
+    const typeset = shadowRoot?.querySelector<HTMLElement>(TYPESET);
+    if (!typeset) return;
+    typeset.style.fontSize = `${fontSize}px`;
+  }, [shadowRoot, fontSize]);
+
+  const handleZoomIn = useCallback(() => {
+    setFontSize(fontSize + 1);
+  }, [fontSize, setFontSize]);
+
+  const handleZoomOut = useCallback(() => {
+    setFontSize(fontSize - 1);
+  }, [fontSize, setFontSize]);
+
+  return (
+    <ButtonGroup variant="outlined">
+      <Tooltip title="Text Zoom In" arrow>
+        <Button size="small" onClick={handleZoomIn} aria-label="text zoom in">
+          <ZoomInIcon />
+        </Button>
+      </Tooltip>
+      <Tooltip title="Text Zoom Out" arrow>
+        <Button size="small" onClick={handleZoomOut} aria-label="text zoom out">
+          <ZoomOutIcon />
+        </Button>
+      </Tooltip>
+    </ButtonGroup>
+  );
+};
+
+export const TextSize = techdocsPlugin.provide(
+  createTechDocsAddonExtension({
+    name: 'TextSize',
+    location: TechDocsAddonLocations.Subheader,
+    component: TextSizeAddon,
+  }),
+);

```

**Review by iamEAP — CHANGES_REQUESTED** [review]
@Rugvip

File: plugins/techdocs/src/Router.tsx (hunk 2 above)
Comment (Rugvip) [reply plugins/techdocs/src/Router.tsx:72]: Indeed, fixed! There was a bit more wrapping than the element discovery was able to handle. Consider switching to [useElementFilter](https://backstage.io/docs/reference/core-plugin-api.useelementfilter) though. I see there's a bit much custom code when traversing the elements, which makes it a bit fragile.

Comment (iamEAP) [reply plugins/techdocs/src/Router.tsx:72]: Indeed.

**Review by iamEAP — APPROVED** [review]
🕺

**Pull request merged**